src/mlpack/core/data/detect_file_type_impl.hpp

```diff
@@ -306,34 +288,36 @@ inline FileType AutoDetect(std::fstream& stream, const std::string& filename)
  * @param filename Name of the file whose type we should detect.
  * @return Detected type of file.
  */
-inline FileType DetectFromExtension(const std::string& filename)
+template<typename DataOptionsType>
+void DetectFromExtension(const std::string& filename,
+                         DataOptionsType& opts)
 {
   const std::string extension = Extension(filename);
 
   if (extension == "csv")
   {
-    return FileType::CSVASCII;
+    opts.Format() = FileType::CSVASCII;
   }
   else if (extension == "txt")
   {
-    return FileType::RawASCII;
+    opts.Format() = FileType::RawASCII;
   }
   else if (extension == "bin")
   {
-    return FileType::ArmaBinary;
+    opts.Format() = FileType::ArmaBinary;
   }
   else if (extension == "pgm")
   {
-    return FileType::PGMBinary;
+    opts.Format() = FileType::PGMBinary;
   }
   else if (extension == "h5" || extension == "hdf5" || extension == "hdf" ||
            extension == "he5")
   {
-    return FileType::HDF5Binary;
+    opts.Format() = FileType::HDF5Binary;
   }
-  else
+  else if (extension == "arff")
   {
-    return FileType::FileTypeUnknown;
+    opts.Format() = FileType::ArffASCII;
   }
```

Comment:
```suggestion
  }
  else
  {
    opts.Format() = FileType::FileTypeUnknown;
  }
```

This will make sure that we still give an error when the type is unknown.